### PR TITLE
Íslandsbanki atm query module

### PR DESCRIPTION
**Pull request by KariSt1**
Query functionality to locate nearest Íslandsbanki atm and get further information about its services.

### CODE REVIEW

File: queries/atm.py
Comment (sultur) [inline queries/atm.py:175]: This (the filtering) can be included in `_get_atm_data()`. Then we can also cache the results so we don't read the file from disk each time a query is performed. I'd make `_get_atm_data` return the `filtered_atm_data` variable, and then add the following decorator:
```py
import cachetools

...

_TTL = 86400 # 1 day

@cachetools.cached(cachetools.TTLCache(1, _TTL))
def _get_atm_data():
    ...
```
This caches the ATM locations in memory for one day (86400 seconds).
We still need to do the distance calculations each time a query is made though.

File: queries/atm.py
Comment (sultur) [inline queries/atm.py:267]: These three lines are repeated in lines 211-213 & 247-249, couldn't we just include the `sorted(atms, ...)` call as the first step in `_group_closest_based_on_address()`?

File: queries/atm.py
Comment (sultur) [inline queries/atm.py:290]: Ah, I think I forgot to mention the code we have for transcribing stuff for TTS in `speech/trans`, like the `gssml` function (you can see usage of it in e.g. the arithmetic module), but I notice now that we haven't implemented a method for transcribing addresses yet. This looks good for the time being though :).

File: queries/atm.py
Comment (sultur) [inline queries/atm.py:324]: Could you move the TODO comments to the top of the file, between the module docstring and the imports? Handy for quickly assessing what needs to be done in each query module (usually a lot of things).

File: queries/atm.py
Comment (sultur) [inline queries/atm.py:354]: I cannot wait for the `match-case` statement introduced in Python 3.10!
Nice job keeping the `if-elif` statements compact and understandable :)
(They can become quite unwieldy in the query modules due to icelandic declensions)

File: queries/atm.py
Comment (sultur) [inline queries/atm.py:521]: _Very minor point:_ It may be a good idea to make the qkeys referenced in this function constants, and reference those (also in the corresponding nonterminal functions). Just decreases the likelihood of minor typos suddenly eradicating functionality from a query module (those can be a joy to debug :smiling_face_with_tear:).

File: queries/atm.py
Comment (sultur) [inline queries/atm.py:341]: Couldn't we just pass in the location as an argument instead of `q: Query`? We already check `q.location` in the `sentence(...)` function.

**Review by sultur — CHANGES_REQUESTED** [review]
Great work on this module!

File: queries/atm.py
Comment (sveinbjornt) [reply queries/atm.py:50]: Absolutely use Pathlib. I'm in the process of moving most of the geo.py code (which uses files in `resources/geo`) into the `iceaddr` external package so this directory might not be around for much longer. I suggest creating a `queries/resources` directory for resources used exclusively by query modules and not elsewhere in the codebase.

File: queries/atm.py
Comment (sveinbjornt) [inline queries/atm.py:63]: This is function is not used unless you define TOPIC_LEMMAS like in some of the other modules. Maybe best to leave it out for now.

File: queries/atm.py
Comment (sveinbjornt) [reply queries/atm.py:175]: Great minds think alike.

**Comment by sveinbjornt** [comment]
This looks good to me, but you should write some tests for this in `tests/test_queries.py`

File: queries/atm.py
Comment (sveinbjornt) [reply queries/atm.py:63]: Or add in the TOPIC_LEMMA "hraðbanki" and more examples of queries supported by this module in random help_text

File: queries/atm.py
Comment (KariSt1) [reply queries/atm.py]: Fixed.

File: queries/atm.py
Comment (KariSt1) [reply queries/atm.py]: Fixed, now using pathlib.

File: queries/atm.py
Comment (KariSt1) [reply queries/atm.py]: Fixed. Added caching to this function and added filtering to it so that it only keeps the atms from the file.

File: queries/atm.py
Comment (KariSt1) [reply queries/atm.py]: Fixed.

File: queries/atm.py
Comment (KariSt1) [reply queries/atm.py]: Fixed. Added constants for the qkeys.

File: queries/atm.py
Comment (KariSt1) [reply queries/atm.py]: Fixed, now passing the location as an argument instead of the query.

File: queries/atm.py
Comment (KariSt1) [reply queries/atm.py:86]: Fixed. Added a topic lemma and further query examples.

File: queries/atm.py
Comment (KariSt1) [reply queries/atm.py]: Fixed.

File: queries/atm.py
Comment (KariSt1) [reply queries/atm.py:300]: Fixed, the sort call is now the first step in `_group_closest_based_on_address()`.

File: queries/atm.py
Comment (KariSt1) [reply queries/atm.py]: Fixed. Longitude, latitude, and street address checks are now made when filtering out the atms in the `get_atm_data` function so there should never be a `KeyError` here anymore.

**Review by sultur — APPROVED** [review]
LGTM, once the tests pass and @sveinbjornt approves we can merge this.

**Pull request merged**
